main.nf

```diff
@@ -736,35 +736,37 @@ process rseqc {
     """
 }
 
-
 /*
- * Step 4.1 Rseqc create BigWig coverage
+ * Step 4.1 Subsample the BAM files if necessary
  */
-
-process createBigWig {
-    tag "${bam.baseName - 'sortedByCoord.out'}"
-    publishDir "${params.outdir}/bigwig", mode: 'copy'
-
-    when:
-    !params.skip_qc && !params.skip_genebody_coverage
+bam_forSubsamp
+    .filter { it.size() > params.subsampFilesizeThreshold }
+    .map { [it, params.subsampFilesizeThreshold / it.size() ] }
+    .set{ bam_forSubsampFiltered }
+bam_skipSubsamp
+    .filter { it.size() <= params.subsampFilesizeThreshold }
+    .set{ bam_skipSubsampFiltered }
+
+process bam_subsample {
+    tag "${bam.baseName - '.sorted'}"
 
     input:
-    file bam from bam_for_genebody
-    file index from bam_index_genebody
+    set file(bam), val(fraction) from bam_forSubsampFiltered
 
     output:
-    file "*.bigwig" into bigwig_for_genebody
+    file "*_subsamp.bam" into bam_subsampled
 
     script:
     """
-    bamCoverage -b $bam -p ${task.cpus} -o ${bam.baseName}.bigwig
+    samtools view -s $fraction -b $bam | samtools sort -o ${bam.baseName}_subsamp.bam
```

Comment:
Very clean! 😀 

Should we move the `samtools index` command here as well? Seems more logical to keep it in this process rather than in the gene body coverage process.

Reply:
Ah no, needs to be there in case we don't do the subsampling 👍